### PR TITLE
CRNS-261: Moved all picker functions to MessageInputContext

### PR DESCRIPTION
**Pull request by vishalnarkhede**
# Submit a pull request

## CLA

- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [ ] The code changes follow best practices
- [ ] Code changes are tested (add some information if not applicable)

## Description of the pull request

This is a PR on top of https://github.com/GetStream/stream-chat-react-native/pull/572

Moved following functions to `MessageInputContext`, to avoid prop drilling on `InputButtons` and `Input` component

- openFilePicker
- openAttachmentPicker
- closeAttachmentPicker
- toggleAttachmentPicker
- openCommandsPicker
- openMentionsPicker

### CODE REVIEW

File: src/contexts/messageInputContext/MessageInputContext.tsx [hunk 1]
```diff
@@ -429,6 +440,13 @@ export const MessageInputProvider = <
   value: InputMessageInputContextValue<At, Ch, Co, Ev, Me, Re, Us>;
 }>) => {
   const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
+  const {
+    closePicker,
+    openPicker,
+    selectedPicker,
+    setSelectedPicker,
+  } = useAttachmentPickerContext();
```
Comment (virdesai) [inline src/contexts/messageInputContext/MessageInputContext.tsx:448]: just a small reordering 😬 
```suggestion
  const {
    closePicker,
    openPicker,
    selectedPicker,
    setSelectedPicker,
  } = useAttachmentPickerContext();

    const { client } = useChatContext<At, Ch, Co, Ev, Me, Re, Us>();
```

File: src/components/MessageInput/MessageInput.tsx [hunk 2]
```diff
@@ -259,22 +256,16 @@ export const MessageInputWithContext = <
...
   } = useAttachmentPickerContext();
 
   useEffect(() => {
     setMaxNumberOfFiles(maxNumberOfFiles ?? 10);
 
-    return () => {
-      closePicker();
-      setSelectedPicker(undefined);
-    };
+    return closeAttachmentPicker;
```
Comment (virdesai) [inline src/components/MessageInput/MessageInput.tsx:268]: hmmm maybe check with @nhannah if there was a reason the order was reversed in the cleanup of this effect vs just using the `closeAttachmentPicker` function which has the calls in the inverse order

File: src/components/MessageInput/MessageInput.tsx [hunk 3]
```diff
@@ -182,19 +182,17 @@ export const MessageInputWithContext = <
 ) => {
   const {
     additionalTextInputProps,
-    appendText,
```
Comment (virdesai) [inline src/components/MessageInput/MessageInput.tsx:185]: with props like this that were removed as they're not used in this component directly anymore, also remove them from the types that are on these functions. For instance, I see `pickFile` below was removed on line 208 but still exists on the type on line 148. Don't want to allow that to be passed in as a type if this component isn't going to be using it

Comment (vishalnarkhede) [reply src/components/MessageInput/MessageInput.tsx:185]: Done!! Have cleaned up few more types, missed from previous PRs :)

**Review by virdesai — APPROVED** [review]
LGTM other than checking with Neil about that other comment I made

File: src/components/MessageInput/MessageInput.tsx (hunk 2 above)
Comment (vishalnarkhede) [reply src/components/MessageInput/MessageInput.tsx:260]: Checked with him :)

**Pull request merged**